### DISCUSSION TITLE
Schema keys order

### DISCUSSION THREAD

**Discussion by jefffairson** [vueform/vueform#367] [Questions]
Hello,

I have a tricky issue that I can't solve...
For a project, multiple form schema are store in a DB in Json as I use the same component for multiple forms to interact with on my page, a quick example of what is stored is this :
```
  "form_title": "test",
  "questions": {
    "type": "group",
    "schema": {...}
  }
``` 

I use Pinia and in my state, objects are stored correctly, this is a screenshot :
![image](https://github.com/user-attachments/assets/5bce0230-960e-45ce-98ea-407245ef9c01)

In my component, the order is respected as well, here is a screenshot :
![image](https://github.com/user-attachments/assets/94566559-6e51-4640-b9be-c10c2792af73)

But on render, ```question``` is displayed first and ```form_title``` comes after. Furthermore, in ```questions``` I have a radiogroup and all items are not sorted as well (but they are in the pinia state and in the component).

I noticed that in the ```refs``` on ```form$```props the ```schema```value is not sorted anymore, here is a screenshot of what happens :
![image](https://github.com/user-attachments/assets/230b3424-8abd-4602-88b9-1c2a6833e066)

My question is, is that possible to keep the keys sorted ? did I miss something in the documentation ? Did this happened to someone else ?

Thanks for your help !
Cheers,
jeFFF

**Comment by adamberecz** [comment]
Unfortunately when using objects (as Vueform does for `schema`) the key order isn't guaranteed. If you add items in a certain order, it will be kept in most of the cases, but if you eg. store them as JSON the DB engine might mix up the order. I'm not 100% sure if `JSON.parse` doesn't do that (it likely does) but probably the first thing I'd try is storing it as a string instead of JSON type and parsing it for rendering.

If that doesn't work either you'll need a converter that transforms any schema object into an array format and back. If you decide to take that direction and need help with the implementation, let me know.
